src/Symfony/Component/Process/Process.php

```diff
@@ -1123,12 +1114,14 @@ private function captureExitCode()
     /**
      * Closes process resource, closes file handles, sets the exitcode.
      *
+     * @param bool $wait Whether to wait for the end of the process or not.
+     *
      * @return int The exitcode
      */
-    private function close()
+    private function close($wait)
     {
         $this->processPipes->close();
-        if (is_resource($this->process)) {
+        if ($wait && is_resource($this->process)) {
             $exitcode = proc_close($this->process);
```

Comment:
could be, but only in cases where sending SIGTERM + SIGKILL did not have any effect, which means the Process object is stuck is an infinite loop.